### DISCUSSION TITLE
Monitor Method And Memory Metrics Of Method Call

### DISCUSSION THREAD

**Discussion by grassbl8d** [open-telemetry/opentelemetry-java-instrumentation#7615] [Q&A]
Hello,

Is there anyway that I can gather Method and Memory metrics of method call ? 
In our current APM tool, this is gathered. 

For example : method call 1 - 100 CPU, Memory Usage - 280000 KB

How do I gather this  ? This is more a trace type metric rather than the general JVM metric.

Thanks,

**Comment by trask** [comment]
hi @grassbl8d! not currently, at least not out-of-the-box

you could capture this using https://docs.oracle.com/en/java/javase/11/docs/api/jdk.management/com/sun/management/ThreadMXBean.html and add that data as attributes onto the span for that method (or as metric attributes)

**Reply by grassbl8d** [reply]
Cool! Thanks. I'll try to look for a feature request for this andnlog one if itgeres none. This would be very useful as others  have it. 

Thanks so much!
